### PR TITLE
Add colors to log module

### PR DESCRIPTION
**Pull request by sh7dm**
Add different colors to log messages by levels

### CODE REVIEW

**Comment by sh7dm** [comment]
Maybe add emojis too?

File: log/handlers.ts [hunk 1]
```diff
@@ -51,6 +52,42 @@ export class BaseHandler {
...
+      const value = logRecord[p1];
+
+      // do not interpolate missing values
+      if (!value) {
+        return match;
+      }
+
+      return value;
+    });
+
```
Comment (bartlomieju) [inline log/handlers.ts]: No need to copy this logic. You can call `super` first and process return value.

File: log/handlers.ts [hunk 2]
```diff
@@ -51,6 +52,42 @@ export class BaseHandler {
 }
 
 export class ConsoleHandler extends BaseHandler {
```
Comment (bartlomieju) [inline log/handlers.ts]: It'd be better if one could enable/disable coloring. Check out `FileHandlerOptions` for example of additional handler config.

Comment (ry) [reply log/handlers.ts]: I agree having that switch would be good - but maybe that should be done in the color module?

Comment (bartlomieju) [reply log/handlers.ts]: Can you give an example?

My idea was:
`new ConsoleHandler("DEBUG", { colors: false })`

**Comment by sh7dm** [comment]
Screenshot:
default

File: log/handlers.ts (hunk 2 above)
Comment (sh7dm) [reply log/handlers.ts]: @ry @bartlomieju done!

File: log/handlers.ts [hunk 3]
```diff
@@ -50,7 +51,46 @@ export class BaseHandler {
   async destroy() {}
 }
 
+interface ConsoleHandlerOptions extends HandlerOptions {
+  colors: boolean;
```
Comment (bartlomieju) [inline log/handlers.ts]: it should be optional field: `colors?: boolean;`

File: log/handlers.ts [hunk 4]
```diff
@@ -50,7 +51,46 @@ export class BaseHandler {
...
+}
+
 export class ConsoleHandler extends BaseHandler {
+  colors = true;
+
+  constructor(levelName: string, options?: ConsoleHandlerOptions) {
+    super(levelName);
```
Comment (bartlomieju) [inline log/handlers.ts]: You missed `options` parameters here: `super(levelName, options)`

File: log/handlers.ts [hunk 5]
```diff
@@ -50,7 +51,46 @@ export class BaseHandler {
...
+
+    if (options && !options.colors) {
```
Comment (bartlomieju) [inline log/handlers.ts]: This conditional will disable colors if they are not explicitly set to `true` in options. I think proper way would be:
`options && options.colors === false`

**Comment by ry** [comment]
We should support https://no-color.org/ in core - I think that's where the logic belongs for switching off color.

**Comment by bartlomieju** [comment]
@ry so should colors be always enabled unless there's `NO_COLOR` env variable? Without option to switch it off without `NO_COLOR`?

**Comment by ry** [comment]
@bartlomieju I think we don't want to require people to use `--allow-env` just to use color... so we should add `deno.NO_COLOR` (or something) which is set at startup... 
Ref https://github.com/denoland/deno/issues/1706

**Comment by bartlomieju** [comment]
@ry then that definitely makes sense to control it directly from `colors` module :+1:

**Comment by bartlomieju** [comment]
@sh7dm I think you can remove `ConsoleHandlerOptions` and check regarding colors. It is now controlled by `NO_COLOR` env variable directly from `colors` - see #182

**Pull request merged**